ci/dependency-update/Jenkinsfile

```diff
@@ -113,18 +117,45 @@ pipeline {
 			post {
 				cleanup {
 					sh 'ci/docker-cleanup.sh'
+					dir('hibernate-orm-local-copy') {
+						deleteDir()
+					}
 				}
 			}
-			steps {
-				// The timeout cannot be in stage options, because that would
-				// include the time needed to provision a node.
-				timeout(time: 30, unit: 'MINUTES') {
-					withMavenWorkspace {
-						sh """ \
+			stages {
+				stage('Build Hibernate ORM') {
+					when {
+						expression {
+							return params.ORM_REPOSITORY?.trim() || params.ORM_BRANCH?.trim()
+						}
+					}
+					steps {
+						script {
+							if (!params.ORM_REPOSITORY?.trim() || !params.ORM_BRANCH?.trim()) {
+								error "Both ORM_REPOSITORY and ORM_BRANCH must be not blank if a local build of Hibernate ORM is required. Repository: [${params.ORM_REPOSITORY}], branch: [${params.ORM_BRANCH}]."
+							}
+						}
+						script {
+							dir('hibernate-orm-local-copy') {
+								sh "git clone ${params.ORM_REPOSITORY} --depth 1 --branch ${params.ORM_BRANCH} --single-branch ."
+								sh "./gradlew publishToMavenLocal -x test -Dmaven.repo.local=${env.WORKSPACE_TMP}/.m2repository"
+							}
+						}
+					}
+				}
+				stage('Build Hibernate Search') {
+					steps {
+						// The timeout cannot be in stage options, because that would
+						// include the time needed to provision a node.
+						timeout(time: 30, unit: 'MINUTES') {
+							withMavenWorkspace {
+								sh """ \
 							mvn clean install -U -Pdist -DskipTests \
 						"""
-						dir(env.WORKSPACE_TMP + '/.m2repository') {
-							stash name: 'original-build-result', includes:"org/hibernate/search/**"
+								dir(env.WORKSPACE_TMP + '/.m2repository') {
+									stash name: 'original-build-result', includes: "org/hibernate/**"
```

Comment:
From what I understand the Hibernate Search build will not use the Hibernate ORM artifacts we just built, which is good. But it will download another version of ORM, and this version will end up in the stash, too...

Do you think it would be possible to just have both builds happen on different nodes, creating strictly separate stashes with strictly separate include patterns?

If I dared, I'd mention you could even have the builds run in parallel :D